checks/__init__.py

```diff
@@ -564,6 +586,14 @@ def run(self):
                     tb=traceback.format_exc()
                 )
             instance_statuses.append(instance_status)
+            # Generate empty metadata if nonexistent
+            service_metadata_count = len(self.service_metadata)
+            if service_metadata_count > i + 1:
```

Comment:
I see, i don't really like the fact that it prevents from calling 
```python
self.svc_metadata
```
more than once per instance. This is bad pattern i think.